src/tests/ftest/io/llnl_mpi4py.py

```diff
@@ -0,0 +1,173 @@
+                                                   self.hostlist_servers,
+                                                   self.hostlist_clients)
+        # start servers
+        server_utils.run_server(self.hostfile_servers, self.server_group,
+                              self.basepath)
+        try:
+            # parameters used in pool create
+            createmode = self.params.get("mode", '/run/pool/createmode/*/')
+            createuid = os.geteuid()
+            creategid = os.getegid()
+            createsetid = self.params.get("setname", '/run/pool/createset/')
+            createsize = self.params.get("size", '/run/pool/createsize/')
+            self.createsvc = self.params.get("svcn", '/run/pool/createsvc/')
+
+            # initialize a python pool object then create the underlying
+            # daos storage
+            self.pool = DaosPool(self.context)
+            self.pool.create(createmode, createuid, creategid,
+                             createsize, createsetid, None, None, self.createsvc)
+        except (DaosApiError) as excep:
+            self.fail("<{0} Test Failed at pool create> \n{1}".format(excep))
+
+    def tearDown(self):
+        try:
+            if self.pool is not None:
+                self.pool.destroy(1)
+        finally:
+            if self.agent_sessions:
+                AgentUtils.stop_agent(self.hostlist_clients,
+                                      self.agent_sessions)
+        server_utils.stop_server(hosts=self.hostlist_servers)
+
+    def run_test(self, test_repo, test_name):
+        """
+        Executable function to be used by test functions below
+        test_repo       --location of test repository
+        test_name       --name of the test to be run
+        """
+        # initialize MpioUtils
+        self.mpio = MpioUtils()
+        if not self.mpio.mpich_installed(self.hostlist_clients):
+            self.fail("Exiting Test: Mpich not installed")
+
+        try:
+            # initialise test specific variables
+            client_processes = self.params.get("np", '/run/client_processes/')
+
+            # obtaining pool uuid and svc list
+            pool_uuid = self.pool.get_uuid_str()
+            svc_list = ""
+            for i in range(self.createsvc):
+                svc_list += str(int(self.pool.svc.rl_ranks[i])) + ":"
+            svc_list = svc_list[:-1]
+
+            # running tests
+            self.mpio.run_llnl_mpi4py(self.basepath, self.hostfile_clients,
+                                      pool_uuid, test_repo, test_name, client_processes)
+
+            # Parsing output to look for failures
+            # stderr directed to stdout
+            stdout = self.logdir + "/stdout"
+            searchfile = open(stdout, "r")
+            error_message = ["non-zero exit code", "MPI_Abort", "MPI_ABORT",
+                             "ERROR"]
+
+            for line in searchfile:
+                for i in range(len(error_message)):
```

Comment:
(pylint-consider-using-enumerate) Consider using enumerate instead of iterating with range and len